AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/UpdateMFAPreferenceTask.swift

```diff
@@ -61,10 +61,12 @@ class UpdateMFAPreferenceTask: AuthUpdateMFAPreferenceTask, DefaultLogger {
 
     func updateMFAPreference(with accessToken: String) async throws {
         let userPoolService = try userPoolFactory()
+        let currentPreference = try await userPoolService.getUser(input: .init(accessToken: accessToken))
+        let preferredMFAType = MFAType(rawValue: currentPreference.preferredMfaSetting ?? "")
```

Comment:
For this transformation, you have to go from `String?` --> `String` --> `MFAType?`.
Using `map` here more accurately represents the transformation because it removes the need for a default value (`preferredMFASetting ?? ""`). 

This may not seem like a big deal now because there's currently no way for `""` to represent a valid `MFAType`. This could be problematic if we add our own `MFAType.init?(rawValue:)` in the future instead of using the synthesized one.

```suggestion
        let preferredMFAType = currentPreference.preferredMfaSetting.map(MFAType.init(rawValue:))
```